### PR TITLE
Allow tests to run against URLs not matching "nuget.org" pattern

### PR DESCRIPTION
**Pull request by joelverhagen**
Addresses https://github.com/NuGet/NuGetGallery/issues/3483.

When testing the staging instance before deploying the gallery, the URL being tested is some random string (dot) cloudapp (dot) .net. This URL does not match the SSL certificate so nuget.exe rejects pushing.

This change adds a custom build of nuget.exe based off of this branch:
https://github.com/NuGet/NuGet.Client/tree/dev-jver-trust-https

This nuget.exe has a feature to trust the fingerprint (a.k.a. thumbprint) of specific certificates. 

In short, this PR:

1. Uses this custom nuget.exe for functional tests.
1. Removes code that was causing some P0 tests to no-op (!!!).
1. Fix one test that depended on a package being in the first page of search results.

As a side note, the feature request for nuget.exe supporting this certificate trust option is here:
https://github.com/NuGet/Home/issues/4387

### CODE REVIEW

File: tests/NuGetGallery.FunctionalTests.Core/EnvironmentSettings.cs [hunk 1]
```diff
@@ -213,5 +216,40 @@ public static string TestEmailServerHost
...
+                {
+                    var unparsedValued = Environment.GetEnvironmentVariable("TrustedHttpsCertificates") ?? string.Empty;
+
+                    List<string> pieces;
+                    if (unparsedValued.Length == 0)
+                    {
+                        pieces = new List<string>
+                        {
+                            "8c11c16610b7a147d10bbcc6a65ce23d321c12c2", // *.nugettest.org
+                            "9d984f91f40d8b3a1fb29153179415523c4e64d1", // *.int.nugettest.org
```
Comment (skofman1) [inline tests/NuGetGallery.FunctionalTests.Core/EnvironmentSettings.cs]: Can you please add a comment about why this is here, and that it will brake if certs are renewed

File: tests/NuGetGallery.FunctionalTests.Core/EnvironmentSettings.cs [hunk 2]
```diff
@@ -213,5 +216,40 @@ public static string TestEmailServerHost
...
+                            "3751cb513b93ee67ec9f18a1f2aec1eac87af9bc"  // *.nuget.org
+                        };
+                    }
+                    else
+                    {
+
```
Comment (skofman1) [inline tests/NuGetGallery.FunctionalTests.Core/EnvironmentSettings.cs]: extra line

File: tests/NuGetGallery.FunctionalTests.Core/Helpers/CommandlineHelper.cs [hunk 3]
```diff
@@ -156,7 +156,13 @@ public CommandlineHelper(ITestOutputHelper testOutputHelper)
             var nugetProcess = new Process();
             var pathToNugetExe = Path.Combine(Environment.CurrentDirectory, NugetExePath);
 
+            foreach (var trustedCertificate in EnvironmentSettings.TrustedHttpsCertificates)
+            {
+                arguments.AddRange(new[] { "-TrustedHttpsCertificate", trustedCertificate });
```
Comment (scottbommarito) [inline tests/NuGetGallery.FunctionalTests.Core/Helpers/CommandlineHelper.cs]: Not a big deal, but I'd expect to see a single semicolon-delimited list rather than a series of `-TrustedHttpsCertificate` arguments...

```
-TrustedHttpsCertificate thumb1;thumb2;thumb3
```
rather than
```
-TrustedHttpsCertificate thumb1 -TrustedHttpsCertificate thumb2 -TrustedHttpsCertificate thumb3
```

File: tests/Scripts/DownloadLatestNuGetExeRelease.ps1 [hunk 4]
```diff
@@ -1,4 +1,11 @@
...
+$sourceNuGetExePath = Join-Path $PSScriptRoot "nuget.exe"
+$targetNugetExePath = ".\nuget.exe"
 
-Invoke-WebRequest $sourceNugetExe -OutFile $targetNugetExe
\ No newline at end of file
+if (-Not (Test-Path $targetNugetExePath)) {
+    if (Test-Path $sourceNuGetExePath) {
+        Copy-Item $sourceNuGetExePath $targetNugetExePath
+    } else {
+        Invoke-WebRequest $sourceNugetExeUrl -OutFile $targetNugetExePath
```
Comment (scottbommarito) [inline tests/Scripts/DownloadLatestNuGetExeRelease.ps1]: maybe should add `-UseBasicParsing`

File: tests/NuGetGallery.FunctionalTests.Core/EnvironmentSettings.cs (hunk 1 above)
Comment (joelverhagen) [inline tests/NuGetGallery.FunctionalTests.Core/EnvironmentSettings.cs]: Fixed.

File: tests/NuGetGallery.FunctionalTests.Core/EnvironmentSettings.cs (hunk 2 above)
Comment (joelverhagen) [inline tests/NuGetGallery.FunctionalTests.Core/EnvironmentSettings.cs]: Fixed.

File: tests/Scripts/DownloadLatestNuGetExeRelease.ps1 (hunk 4 above)
Comment (joelverhagen) [inline tests/Scripts/DownloadLatestNuGetExeRelease.ps1]: `-UseBasicParsing` only applies to HTML responses. This is a binary.

File: tests/NuGetGallery.FunctionalTests.Core/Helpers/CommandlineHelper.cs (hunk 3 above)
Comment (joelverhagen) [inline tests/NuGetGallery.FunctionalTests.Core/Helpers/CommandlineHelper.cs]: Yeah, that would be more terse, but per the feature request mentioned above, I will likely configure this setting in NuGet.config and maybe an environment variable that nuget.exe expects.

File: tests/NuGetGallery.FunctionalTests.Core/EnvironmentSettings.cs [hunk 5]
```diff
@@ -213,5 +216,42 @@ public static string TestEmailServerHost
...
+                    if (unparsedValued.Length == 0)
+                    {
+                        // This list will need to be modified as DEV, INT, and PROD certificates change and are
+                        // renewed. These values are easily and publicly discoverable by inspecting the certificate
+                        // returned from HTTPS browser interactions with the gallery.
+                        pieces = new List<string>
+                        {
+                            "8c11c16610b7a147d10bbcc6a65ce23d321c12c2", // *.nugettest.org
+                            "9d984f91f40d8b3a1fb29153179415523c4e64d1", // *.int.nugettest.org
+                            "3751cb513b93ee67ec9f18a1f2aec1eac87af9bc"  // *.nuget.org
```
Comment (shishirx34) [inline tests/NuGetGallery.FunctionalTests.Core/EnvironmentSettings.cs]: This will need to be changed for prod deployment, it will likely fail for prod staging functional tests, because it will use the new cert.

Comment (joelverhagen) [inline tests/NuGetGallery.FunctionalTests.Core/EnvironmentSettings.cs]: Yes, intentional. I would like to exercise this test failure case. Thanks for the heads up.
